channels/channels.go

```diff
@@ -0,0 +1,87 @@
+package channels
```

Comment:
Moved `channels` into it's own package because it is a huge package. I can see this growing even more making it harder to move in the future, so lets do it now.